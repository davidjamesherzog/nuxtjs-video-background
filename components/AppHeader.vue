<template>
  <header>
    <h2 class="logo" data-app-header-logo>video.js</h2>
    <div
      class="toggle"
      :class="{ active: active }"
      data-app-header-toggle
      @click="toggleMenu"
    ></div>
  </header>
</template>

<script>
export default {
  data() {
    return {
      active: false
    };
  },
  methods: {
    toggleMenu() {
      this.active = !this.active;
      this.$emit('menu');
    }
  }
};
</script>

<style lang="scss" scoped>
header {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  padding: 40px 100px;
  z-index: 1000;
  display: flex;
  justify-content: space-between;
  align-items: center;

  .logo {
    color: $color-brand-white;
    text-transform: uppercase;
    cursor: pointer;
  }

  .toggle {
    position: relative;
    width: 60px;
    height: 60px;
    background: url('../assets/images/menu.png');
    background-repeat: no-repeat;
    background-size: 30px;
    background-position: center;
    cursor: pointer;

    &.active {
      background: url('../assets/images/close.png');
      background-repeat: no-repeat;
      background-size: 25px;
      background-position: center;
      cursor: pointer;
    }
  }
}

@media (max-width: 991px) {
  header {
    padding: 40px;
  }
}
</style>
