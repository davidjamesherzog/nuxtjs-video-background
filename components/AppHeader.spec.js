import { mount } from '@vue/test-utils';
import AppHeader from '@/components/AppHeader.vue';

describe('AppHeader', () => {
  describe('data', () => {
    test('returns defaults', () => {
      expect(AppHeader.data()).toEqual({
        active: false
      });
    });
  });

  describe('methods', () => {
    describe('toggleMenu', () => {
      test('should set active to true', () => {
        const mockContext = {
          active: false,
          $emit: jest.fn()
        };
        AppHeader.methods.toggleMenu.bind(mockContext)();
        expect(mockContext.active).toBeTruthy();
        expect(mockContext.$emit).toHaveBeenCalledWith('menu');
      });

      test('should set active to false', () => {
        const mockContext = {
          active: true,
          $emit: jest.fn()
        };
        AppHeader.methods.toggleMenu.bind(mockContext)();
        expect(mockContext.active).toBeFalsy();
        expect(mockContext.$emit).toHaveBeenCalledWith('menu');
      });
    });
  });

  describe('templates', () => {
    test('is a Vue instance', () => {
      const wrapper = mount(AppHeader);
      expect(wrapper.vm).toBeTruthy();
      expect(wrapper.element).toMatchSnapshot();
    });

    test('should not add active class', () => {
      const wrapper = mount(AppHeader, {
        data: () => {
          return {
            active: false
          };
        }
      });
      const logo = wrapper.find('[data-app-header-logo]');
      const toggle = wrapper.find('[data-app-header-toggle]');

      expect(logo.exists()).toBeTruthy();
      expect(logo.classes()).toContain('logo');
      expect(logo.text()).toContain('video.js');

      expect(toggle.exists()).toBeTruthy();
      expect(toggle.classes()).toContain('toggle');
      expect(toggle.classes()).not.toContain('active');
      expect(wrapper.element).toMatchSnapshot();
    });

    test('should add active class', () => {
      const wrapper = mount(AppHeader, {
        data: () => {
          return {
            active: true
          };
        }
      });
      const logo = wrapper.find('[data-app-header-logo]');
      const toggle = wrapper.find('[data-app-header-toggle]');

      expect(logo.exists()).toBeTruthy();
      expect(logo.classes()).toContain('logo');
      expect(logo.text()).toContain('video.js');

      expect(toggle.exists()).toBeTruthy();
      expect(toggle.classes()).toContain('toggle');
      expect(toggle.classes()).toContain('active');
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
