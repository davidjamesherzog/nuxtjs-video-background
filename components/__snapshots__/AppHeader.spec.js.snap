// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader templates is a Vue instance 1`] = `
<header>
  <h2
    class="logo"
    data-app-header-logo=""
  >
    video.js
  </h2>
   
  <div
    class="toggle"
    data-app-header-toggle=""
  />
</header>
`;

exports[`AppHeader templates should add active class 1`] = `
<header>
  <h2
    class="logo"
    data-app-header-logo=""
  >
    video.js
  </h2>
   
  <div
    class="toggle active"
    data-app-header-toggle=""
  />
</header>
`;

exports[`AppHeader templates should not add active class 1`] = `
<header>
  <h2
    class="logo"
    data-app-header-logo=""
  >
    video.js
  </h2>
   
  <div
    class="toggle"
    data-app-header-toggle=""
  />
</header>
`;
